flagd/cmd/start.go

```diff
@@ -18,16 +18,17 @@ import (
 const (
 	corsFlagName           = "cors-origin"
 	logFormatFlagName      = "log-format"
-	metricsExporter        = "metrics-exporter"
 	managementPortFlagName = "management-port"
+	metricsExporter        = "metrics-exporter"
 	otelCollectorURI       = "otel-collector-uri"
 	portFlagName           = "port"
 	serverCertPathFlagName = "server-cert-path"
 	serverKeyPathFlagName  = "server-key-path"
 	socketPathFlagName     = "socket-path"
 	sourcesFlagName        = "sources"
+	syncEnabledFlagName    = "sync-enabled"
```

Comment:
Not sure, but we could _probably_ just remove this flag and always serve this. If you don't want to use it, you can simply not expose it on the network. Interested in other opinions on this one.

Reply:
I added this service as "disabled by default" so that users won't be affected by a new service when we do the next release. This can happen if a K8s deployment uses the latest tag or by mistake if someone forgets to check release notes.

Reply:
> I added this service as "disabled by default" so that users won't be affected by a new service when we do the next release. This can happen if a K8s deployment uses the latest tag or by mistake if someone forgets to check release notes.

That makes sense, however, since this is on a different port, I would be surprised if there was any severe networking or security implications since no forwarding/exposure would be occurring on this port from the wider infra. I think just mentioning this explicitly on the docs and title should be enough to assuage that concern.

I don't feel strongly about this point, but I think it's a bit odd having to opt into this considering we don't require users to opt into the evaluation.proto, and they are both equally valid to use.

My suggestion would be to remove this and change the title to `feat: serve sync.proto on port 8015` which will make things very clear.

Reply:
I will wait for @beeme1mr , @thisthat and @bacherfl feedback and do this change if all of is agree :)

Reply:
I agree, I think if this new endpoint is highlighted explicitly in the release notes, the flag should not be required

Reply:
I like the idea of hiding the new functionalities behind a "release feature flag" :p But I also agree with the points of @toddbaert and I don't think we really need this

Reply:
Done, I have removed the flag with commit https://github.com/open-feature/flagd/pull/1237/commits/1c330ff9dcfcf9fdab4a3a6031b29c50ff3e53f3